File: money/money.py
'''
ID: michael504
LANG: PYTHON3   
TASK: money
'''

fin = open('money.in', 'r')
v, n = map(int, fin.readline().split())
a = []
for i in range(n)
dp = [0 for i in range(10001)]
dp[0] = 1
for i in range(v):
    for j in range(n+1):
        print(i, len(a))
        if (j+a[i] <= n):
            dp[a[i]+j] += dp[j]
with open('money.out', 'w') as f:
        f.write(str(dp[n]) + '\n')

File: money/money.cpp
/*
ID: michael504
LANG: C++
TASK: money
*/
#include <bits/stdc++.h>

using namespace std;
int main()  {

	freopen("money.in", "r", stdin);
	freopen("money.out", "w", stdout);
    int v, n, a[26];
    long long dp[100001];
    cin >> v >> n;
    for (int i = 0; i < v; i++) {
        cin >> a[i];
    }

	dp[0] = 1;
    for (int i = 0; i < v; i++) {
        for (int j = 0; j <= n; j++)    {
            if (a[i] + j <= n)  {
                dp[a[i] + j] += dp[j];
            }
        }
    }
    cout << dp[n] << endl;
}